Library/Homebrew/unpack_strategy.rb

```diff
@@ -61,23 +99,29 @@ def self.from_type(type)
     end
   end
 
+  sig { params(extension: String).returns(T.nilable(AnyStrategy)) }
   def self.from_extension(extension)
-    strategies.sort_by { |s| s.extensions.map(&:length).max || 0 }
-              .reverse
-              .find { |s| s.extensions.any? { |ext| extension.end_with?(ext) } }
+    strategies&.sort_by { |s| s.extensions.map(&:length).max || 0 }
+              &.reverse
+              &.find { |s| s.extensions.any? { |ext| extension.end_with?(ext) } }
   end
 
+  sig { params(path: Pathname).returns(T.nilable(AnyStrategy)) }
   def self.from_magic(path)
-    strategies.find { |s| s.can_extract?(path) }
+    strategies&.find { |s| s.can_extract?(path) }
   end
 
+  sig {
+    params(path: Pathname, prioritize_extension: T::Boolean, type: T.nilable(Symbol), ref_type: T.nilable(String),
+           ref: T.nilable(String), merge_xattrs: T.nilable(T::Boolean)).returns(T.untyped)
+  }
   def self.detect(path, prioritize_extension: false, type: nil, ref_type: nil, ref: nil, merge_xattrs: nil)
     strategy = from_type(type) if type
 
     if prioritize_extension && path.extname.present?
       strategy ||= from_extension(path.extname)
-      strategy ||= strategies.select { |s| s < Directory || s == Fossil }
-                             .find { |s| s.can_extract?(path) }
+      strategy ||= strategies&.select { |s| s < Directory || s == Fossil }
+                             &.find { |s| s.can_extract?(path) }
```

Comment:
and maybe an `if` here

Reply:
This is because `merge_xattrs: nil`.

Reply:
this can/should be done in one pass, right? something like `strategies&.find { |s| (s < Directory || s == Fossil) && s.can_extract?(path) }`

Reply:
Yeh, that seems better.